portal/server/utils/csv-utils.js

```diff
@@ -43,11 +56,31 @@ const parseXlsxToCsvArrays = (worksheet) => {
       rowData.push(cellValue);
       if (firstRow) {
         rowDataWithCellAddresses.push(`${cellValue}`);
+        headerCount += 1;
       } else {
         rowDataWithCellAddresses.push(`${cellValue}-${cell.address}`);
+        lastAddress = cell.address;
+        columnCount += 1;
       }
     });
+
+    if (firstRow === false) {
+      // If the row has no data in the final columns of the row, we need to fill in the empty cells with
+      // empty strings and cell addresses so that the csv parser library can parse the data correctly
+      while (columnCount < headerCount) {
+        rowData.push('');
+        // calculate the cell address here given the last address
+        const lastAddressMatch = lastAddress.match(CELL_ADDRESS_REGEX);
+        const [lastColumn, lastRow] = lastAddressMatch.slice(1);
+        const lastColumnIndex = excelColumnToColumnIndex(lastColumn);
+        const newColumn = columnIndexToExcelColumn(parseInt(lastColumnIndex, 10) + 1);
```

Comment:
If we give a (JSDoc) return type of `excelColumnToColumnIndex` I think we can remove `parseInt` here?